### PR TITLE
Add Text replicate, spaces

### PR DESCRIPTION
**Pull request by ari-becker**
Adds a function which generates Text with the number of spaces specified by the parameter.

I believe that this merits addition to the standard library because, when generating a configuration file with `dhall text`, it is common for configuration files to have significant whitespace (indentation etc.), and adding this function would make it significantly easier for users to generate precise amounts of whitespace in order to match downstream configuration format requirements.

### CODE REVIEW

**Comment by sjakobi** [comment]
The implementation could also be slightly simplified by using `Prelude.List.replicate`.

**Comment by Gabriel439** [comment]
Yeah, I also think a `Text/replicate` function would be great and should be easy to implement in terms of `List/replicate` and `Text/concat`

**Comment by ari-becker** [comment]
Cool, I've incorporated both of your feedback. I did decide to keep `generateWhitespace` (now `spaces`) as a separate function from `replicate` as I think that it's a common-enough use-case to merit inclusion as a distinct function.

File: Prelude/Text/replicate [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        ../List/replicate sha256:d4250b45278f2d692302489ac3e78280acb238d27541c837ce46911ff3baa347
+      ? ../List/replicate
+
+let replicate
+    : Natural → Text → Text
+    = λ(num : Natural) → λ(text : Text) → concat (List/replicate num Text text)
+
+let example0 = assert : replicate 3 "foo" ≡ "foofoofoo"
+
+let example1 = assert : replicate 0 "foo" ≡ ""
```
Comment (Gabriel439) [inline Prelude/Text/replicate:19]:
```suggestion
let property = \(text : Text) -> assert : replicate 0 text ≡ ""
```

**Pull request merged**